interactive/views.py

```diff
@@ -50,8 +44,8 @@ def new_analysis_request(request):
     if request.method == "POST":
         form = AnalysisRequestForm(request.POST, codelists=codelists)
         if form.is_valid():
-            form.save(user=request.user)
-            notify_analysis_request_submitted(form.instance)
+            analysis_request = form.save(user=request.user)
+            run_analysis(analysis_request)
```

Comment:
Basically, I think it might be better to rename run.py to something less generic, because it's likely to become confusing

Reply:
Good question. I spent a while thinking about the fact that it's creating a study definition, but couldn't come up with a good name. Then I realised that at the moment, it's creating a commit and pushing it, the study is kind of separate (and would be moved out if it became more complex). So I would be tempted to call it git.py and move it into services. I'm not too fussed though.

Reply:
Looking ahead, it *will* run it, probably within a month. So I was thinking ahead to that situation (which is why I moved the notify in there, as it would potentially either go away or change when that happens).

IMO, the only thing in run that could go in a services/git.py IMO is the `git()` function. `write_files` is core service logic specific to OSI. And I think the commit metadata is similarly key logic, although I could see a case being made for abstracting it a bit more.